packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1551,6 +1562,25 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Stops all running snaps, removes all hooks, closes all connections, and
+   * terminates their workers.
+   *
+   * @param statusEvent - The Snap status event that caused the snap to be
+   * stopped.
+   */
+  public async stopAllRunningSnaps(
```

Comment:
Seems implied that the Snaps have to be running to stop them 😅

```suggestion
  public async stopAllSnaps(
```

Reply:
Haha I did have it like that originally but then I started to wonder if I was specific enough 😂

Reply:
updated [bb3bd45](https://github.com/MetaMask/snaps/pull/2674/commits/bb3bd452dae71c6fb5512cd945e77835ce38251c)